### PR TITLE
Add entries transformer before indexation

### PR DESCRIPTION
**Pull request by bidoubiwa**
related: 

- #264 
- #249 
- #248 
- #218
- #211 
- #112
- #42

To transform the data of a entry before sending it to MeiliSearch, the transform function must be added in the related model configuration. 

For example a collection `restaurant`  has a config file located at `/api/restaurnt/models/restaurant.js` 

Inside you can add a method named `transformEntry` in the meilisearch object: 

For example to sanitize your data: 
```js
'use strict'
const { sanitizeEntity } = require('strapi-utils')

module.exports = {
  meilisearch: {
    transformEntry(entry, model) {
      entry = {
        ...entry,
        categories: entry.categories.map(cat => cat.name)
      };
      entry =  sanitizeEntity(entry, { model })
      return entry
    },
    indexName: "my_restaurant"
  }
}
```

### CODE REVIEW

**Comment by jbelke** [comment]
@bidoubiwa testing against the larger index on #296 , thank you for the direction.

File: connectors/__tests__/entry-transformer.tests.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      collectionConnector,
+      'getEntriesBatch'
+    )
+
+    const transFormEntriesSpy = jest.spyOn(
+      collectionConnector,
+      'transformEntries'
+    )
+    await meilisearchConnector.addCollectionInMeiliSearch('restaurant')
+    // expect(servicesMock.restaurant.count).toHaveBeenCalledTimes(1)
```
Comment (curquiza) [inline connectors/__tests__/entry-transformer.tests.js:95]: IS there a reason why this line is commented?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -55,10 +55,12 @@ Each PR should pass the tests and the linter to be accepted.
...
 
-# Tests the project
+# Integration tests
 yarn test
-# Tests the project in watch/open mode
-yarn test:watch
+# Tests the admin page in a browser
+yarn test:e2e
+# Tests the admin page in a browser in watch mode
+yarn test:e2e:watch
```
Comment (curquiza) [inline CONTRIBUTING.md:63]: I don't see it in the package.json but maybe I'm wrong.

File: connectors/__tests__/entry-transformer.tests.js (hunk 1 above)
Comment (bidoubiwa) [reply connectors/__tests__/entry-transformer.tests.js:95]: No!

**Review by curquiza — APPROVED** [review]
💪

**Pull request merged**